azure-pipelines.yml

```diff
@@ -62,19 +65,31 @@ jobs:
         condition: eq(variables.type, 'linux')
 
       - script: |
-          GOARCH=amd64 GOOS=windows go build -o "$(Build.ArtifactStagingDirectory)/azcopy_windows_amd64.exe"
+          go build -o "$(Build.ArtifactStagingDirectory)/azcopy_windows_amd64.exe"
         displayName: 'Generate Windows AMD64'
-        condition: eq(variables.type, 'linux')
+        condition: eq(variables.type, 'windows')
+        env:
+          GOARCH: amd64
+          GOOS: windows
 
       - script: |
-          GOARCH=386 GOOS=windows go build -o "$(Build.ArtifactStagingDirectory)/azcopy_windows_386.exe"
+          go build -o "$(Build.ArtifactStagingDirectory)/azcopy_windows_386.exe"
         displayName: 'Generate Windows i386'
-        condition: eq(variables.type, 'linux')
+        condition: eq(variables.type, 'windows')
+        env:
+          GOARCH: 386
+          GOOS: windows
+          CGO_ENABLED: 1
 
       - script: |
-          GOARCH=arm GOARM=7 GOOS=windows go build -o "$(Build.ArtifactStagingDirectory)/azcopy_windows_v7_arm.exe"
+          go build -o "$(Build.ArtifactStagingDirectory)/azcopy_windows_v7_arm.exe"
         displayName: 'Generate Windows ARM'
-        condition: eq(variables.type, 'linux')
+        condition: eq(variables.type, 'windows')
+        env:
+          GOARCH: arm
+          GOARM: 7
+          GOOS: windows
+          CGO_ENABLED: 1
```

Comment:
please disable this once you investigate on the ieproxy side! other than that LGTM!